aws/table_aws_kinesis_consumer.go

```diff
@@ -0,0 +1,149 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsKinesisConsumer(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_kinesis_consumer",
+		Description: "AWS Kinesis Consumer",
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.SingleColumn("consumer_arn"),
+			ShouldIgnoreError: isNotFoundError([]string{"ResourceNotFoundException"}),
+			Hydrate:           getAwsKinesisConsumer,
+		},
+		List: &plugin.ListConfig{
+			ParentHydrate: listStreams,
+			Hydrate:       listKinesisConsumer,
+		},
+		GetMatrixItem: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "consumer_name",
+				Description: "The name of the consumer is something you choose when you register the consumer.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "consumer_arn",
+				Description: "When you register a consumer, Kinesis Data Streams generates an ARN for it.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ConsumerARN"),
+			},
+			{
+				Name:        "stream_arn",
+				Description: "When you register a consumer, Kinesis Data Streams generates an ARN for it.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getAwsKinesisConsumer,
+				Transform:   transform.FromField("StreamARN"),
+			},
+			{
+				Name:        "consumer_status",
+				Description: "A consumer can't read data while in the CREATING or DELETING states.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "consumer_creation_timestamp",
+				Description: "The name of the consumer is something you choose when you register the consumer.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			// Standard columns for all tables
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ConsumerName"),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ConsumerARN").Transform(arnToAkas),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listKinesisConsumer(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

Comment:
function name should be `listKinesisConsumers`

Reply:
resolved